apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -680,25 +649,40 @@ defmodule Aecore.Chain.Worker do
       {key = :accounts, root_hash}, acc_state ->
         Map.put(acc_state, key, PatriciaMerkleTree.new(key, root_hash))
 
-      # TODO
-      # This workaround was made until the Oracles were converted to PatriciaMerkleTree #GH-349
+      {_key = :oracles,
+       %{oracle_tree: oracle_root_hash, oracle_cache_tree: oracle_cache_root_hash}},
+      acc_state ->
+        oracle_tree = %{:oracle_tree => PatriciaMerkleTree.new(:oracles, oracle_root_hash)}
+
+        oracle_cache_tree = %{
+          :oracle_cache_tree => PatriciaMerkleTree.new(:oracles_cache, oracle_cache_root_hash)
+        }
+
+        put_in(acc_state, [:oracles], Map.merge(oracle_tree, oracle_cache_tree))
+
       {key, value}, acc_state ->
         Map.put(acc_state, key, value)
     end
   end
 
   defp get_persist_strategy(:from_chainstate) do
     fn
-      {key = :naming, value}, acc_state ->
+      {key = :accounts, value}, acc_state ->
         Map.put(acc_state, key, value.root_hash)
 
-      {key = :accounts, value}, acc_state ->
+      {key = :naming, value}, acc_state ->
         Map.put(acc_state, key, value.root_hash)
 
-      # TODO
-      # This workaround was made until the Oracles were converted to PatriciaMerkleTree #GH-349
+      {key = :oracles, value}, acc_state ->
+        Map.put(acc_state, key, %{
+          oracle_tree: value.oracle_tree.root_hash,
+          oracle_cache_tree: value.oracle_cache_tree.root_hash
+        })
+
       {key, value}, acc_state ->
```

Comment:
@DanielaIvanova this case is not needed anymore, as every chainstate is now handled explicitly, right?

Reply:
I will remove it